src/Sylius/Component/Core/Storage/CartStorageInterface.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Storage;
+
+use Sylius\Component\Core\Model\ChannelInterface;
+
+interface CartStorageInterface
+{
+    /**
+     * @param ChannelInterface $channel
+     *
+     * @return bool
+     */
+    public function hasCartIdForChannel(ChannelInterface $channel): bool;
+
+    /**
+     * @param ChannelInterface $channel
+     *
+     * @return mixed
+     */
+    public function getCartIdForChannel(ChannelInterface $channel);
+
+    /**
+     * @param ChannelInterface $channel
+     * @param mixed $cartId
+     */
+    public function setCartIdForChannel(ChannelInterface $channel, $cartId): void;
+
+    /**
+     * @param ChannelInterface $channel
+     */
+    public function removeCartIdFromChannel(ChannelInterface $channel): void;
```

Comment:
The "from channel" part looks kinda wrong to me, I understand it like it implies that channel has cart it assigned. Why not "for channel" instead - `removeCartIdForChannel`?